phoenix-core/src/it/java/org/apache/phoenix/end2end/SystemCatalogRollbackEnabledIT.java

```diff
@@ -0,0 +1,298 @@
+            }
+
+            // alter child view with drop column should be allowed
+            ddl = "ALTER VIEW " + childViewName + " DROP COLUMN col2";
+            conn.createStatement().execute(ddl);
+        }
+    }
+
+    // Test for PHOENIX-6032
+    @Test
+    public void testViewDoesNotSeeDataForDroppedColumn() throws Exception {
+        final String parentName = "T_" + SchemaUtil.getTableName(
+                generateUniqueName(), generateUniqueName());
+        final String viewName = "V_" + SchemaUtil.getTableName(
+                generateUniqueName(), generateUniqueName());
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.createStatement().execute("CREATE TABLE " + parentName
+                    + " (A INTEGER PRIMARY KEY, B INTEGER, C"
+                    + " VARCHAR, D INTEGER)");
+            conn.createStatement().execute("CREATE VIEW " + viewName
+                    + " (VA INTEGER, VB INTEGER)"
+                    + " AS SELECT * FROM " + parentName + " WHERE B=200");
+            // Upsert some data via the view
+            conn.createStatement().execute("UPSERT INTO " + viewName
+                    + " (A,B,C,D,VA,VB) VALUES"
+                    + " (2, 200, 'def', -20, 91, 101)");
+            conn.commit();
+        }
+
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            // Query from the parent table and assert expected values
+            ResultSet rs = conn.createStatement().executeQuery(
+                    "SELECT A,B,C,D FROM " + parentName);
+            assertTrue(rs.next());
+            assertEquals(2, rs.getInt(1));
+            assertEquals(200, rs.getInt(2));
+            assertEquals("def", rs.getString(3));
+            assertEquals(-20, rs.getInt(4));
+            assertFalse(rs.next());
+
+            // Query from the view and assert expected values
+            rs = conn.createStatement().executeQuery(
+                    "SELECT A,B,C,D,VA,VB FROM " + viewName);
+            assertTrue(rs.next());
+            assertEquals(2, rs.getInt(1));
+            assertEquals(200, rs.getInt(2));
+            assertEquals("def", rs.getString(3));
+            assertEquals(-20, rs.getInt(4));
+            assertEquals(91, rs.getInt(5));
+            assertEquals(101, rs.getInt(6));
+            assertFalse(rs.next());
+        }
+
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            // Drop a parent column from the view
+            conn.createStatement().execute("ALTER VIEW " + viewName
+                    + " DROP COLUMN C");
+            try {
+                conn.createStatement().executeQuery("SELECT C FROM "
+                        + viewName);
+                fail("Expected a ColumnNotFoundException for C since it "
+                        + "was dropped");
+            } catch (ColumnNotFoundException ignore) { }
+        }
```

Comment:
Should also fail if we get some other exception other than ColumnNotFoundException -- can have a second catch block for Throwable and then fail() in it.

Reply:
If any other exception apart from CNFE gets thrown here the test will fail anyways.